DscResources/MSFT_xSmbShare/MSFT_xSmbShare.psm1

```diff
@@ -1,3 +1,641 @@
+
+    
+    for ($i = 0; $i -lt $numbernoaccess; $i++)
+    { 
+        
+      $ErrorActionPreference = "SilentlyContinue"
+
+      $result =  ($CurrentConfiguration.noaccess   -contains $PSBound.NoAccess[$i]).ToString()  
+
+      $user =  $PSBound.NoAccess[$i] 
+      
+      if($user){
+
+## If the User should should have permission we will  add him to this Variable
+
+      $TestingcorrectperrmissionAndparameters += $result
+      $addeduser = $PSBound.NoAccess
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $ErrorActionPreference = "Continue"
+      
+      $user=$null
+      
+      }
+      
+     }
+        
+    }#end IF 
+
+If   ($PSBound.ContainsKey('readaccess')){
+
+     for ($i = 0; $i -lt $numberreadaccess; $i++)
+    { 
+      $ErrorActionPreference = "SilentlyContinue"
+
+     $result =  ($CurrentConfiguration.readaccess   -contains    $PSBound.readaccess[$i]).ToString() 
+
+     $user = $PSBound.readaccess[$i]
+     
+
+      if($user){
+
+
+## If the User should should have permission we will  add him to this Variable
+
+      $TestingcorrectperrmissionAndparameters += $result
+
+      $addeduser = $PSBound.readaccess
+
+      $ErrorActionPreference = "SilentlyContinue"
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $user=$null
+
+      $ErrorActionPreference = "Continue"
+      } 
+
+     }   
+    
+    } #end IF
+
+If   ($PSBound.ContainsKey('fullaccess')){
+
+     for ($i = 0; $i -lt $numberfullaccess ; $i++)
+    { 
+     $ErrorActionPreference = "SilentlyContinue"
+
+     $result =  ($CurrentConfiguration.fullaccess  -contains    $PSBound.fullaccess[$i] ).ToString()  
+
+     $user = $PSBound.fullaccess[$i]
+     
+
+
+     if($user){
+
+
+## If the User should should have permission we will  add him to this Variable
+
+     $TestingcorrectperrmissionAndparameters += $result
+
+      $addeduser = $PSBound.fullaccess
+      $AddingUserPermission.add($addeduser[$i],$result)
+
+      $ErrorActionPreference = "Continue"
+      $user=$null
+
+      }
+
+        
+    }
+
+
+
+
+}#end IF
+
+If   ($PSBound.ContainsKey('ChangeAccess')){
+
+     for ($i = 0; $i -lt $numberChangeAccess; $i++)
+    { 
+     $ErrorActionPreference = "SilentlyContinue"
+     $result =  ($CurrentConfiguration.ChangeAccess   -contains   $PSBound.ChangeAccess[$i]).ToString() 
+
+     $user = $PSBound.ChangeAccess[$i]
+
+ 
+
+     if($user){
+
+      
+## we are adding user to Variable      
+      $TestingcorrectperrmissionAndparameters += $result
+
+      $addeduser = $PSBound.ChangeAccess
+  
+      $AddingUserPermission.add($addeduser[$i],$result)
+      $ErrorActionPreference = "Continue"
+      $user=$null
+      `
+      } 
+
+     }   
+    
+    } #end IF
+ 
+$AddingUserPermissiontoObject = @{}
+
+## Creating Object for with user and permission that needs to be assigned 
+
+$usersadd = ($AddingUserPermission.GetEnumerator() | Where-Object { $_.value -eq "False"}).name 
+
+foreach ( $user in $usersadd){
```

Comment:
Feedback: It's difficult to read this code without proper indentation and with all the extra newlines.